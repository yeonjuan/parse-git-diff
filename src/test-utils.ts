import Context from './context';

type Validator<Args extends any[] = any[]> = (...args: Args) => boolean;

function createBooleanTester<V extends Validator>(
  validator: V,
  result: boolean
) {
  return function validTester(cases: Parameters<V>[]): void {
    cases.forEach((c) => {
      expect(validator(...c)).toBe(result);
    });
  };
}

export function createValidTester<V extends Validator>(validator: V) {
  return createBooleanTester(validator, true);
}

export function createInvalidTester<V extends Validator>(validator: V) {
  return createBooleanTester(validator, false);
}

export function createContext(diff: string, initial: number = 1) {
  const context = new Context(diff);
  for (let i = 0; i < initial - 1; i++) {
    context.nextLine();
  }
  return context;
}
