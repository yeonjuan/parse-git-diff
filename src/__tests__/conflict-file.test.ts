import { getFixture } from './test-utils';
import parseGitDiff from '../parse-git-diff';

describe('conflict-file', () => {
  const fixture = getFixture('conflict-file');

  it('parse `conflict-file`', () => {
    expect(parseGitDiff(fixture)).toMatchSnapshot();
  });
});
