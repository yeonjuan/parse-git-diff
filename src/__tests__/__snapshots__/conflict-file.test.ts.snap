// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conflict-file parse \`conflict-file\` 1`] = `
Object {
  "files": Array [
    Object {
      "chunks": Array [
        Object {
          "changes": Array [
            Object {
              "content": " npm install parse-git-diff",
              "lineAfter": 8,
              "lineBefore": 8,
              "type": "UnchangedLine",
            },
            Object {
              "content": " \`\`\`",
              "lineAfter": 9,
              "lineBefore": 9,
              "type": "UnchangedLine",
            },
            Object {
              "content": " ",
              "lineAfter": 10,
              "lineBefore": 10,
              "type": "UnchangedLine",
            },
            Object {
              "content": "+<<<<<<< HEAD",
              "lineAfter": 11,
              "type": "AddedLine",
            },
            Object {
              "content": "+## a",
              "lineAfter": 12,
              "lineBefore": 11,
              "type": "UnchangedLine",
            },
            Object {
              "content": "+=======",
              "lineAfter": 13,
              "type": "AddedLine",
            },
            Object {
              "content": " ## b",
              "lineAfter": 14,
              "type": "AddedLine",
            },
            Object {
              "content": "+>>>>>>> branch-b",
              "lineAfter": 15,
              "type": "AddedLine",
            },
            Object {
              "content": " ",
              "lineAfter": 16,
              "lineBefore": 12,
              "type": "UnchangedLine",
            },
            Object {
              "content": " - [demo](https://yeonjuan.github.io/parse-git-diff/)",
              "lineAfter": 17,
              "lineBefore": 13,
              "type": "UnchangedLine",
            },
            Object {
              "content": " ",
              "lineAfter": 18,
              "lineBefore": 14,
              "type": "UnchangedLine",
            },
            Object {
              "content": " ",
              "lineAfter": 19,
              "lineBefore": 15,
              "type": "UnchangedLine",
            },
          ],
          "fromFileRangeA": Object {
            "lines": 7,
            "start": 8,
          },
          "fromFileRangeB": Object {
            "lines": 7,
            "start": 8,
          },
          "toFileRange": Object {
            "lines": 11,
            "start": 8,
          },
          "type": "CombinedChunk",
        },
      ],
      "path": "README.md",
      "type": "ChangedFile",
    },
  ],
  "type": "GitDiff",
}
`;
